listenbrainz/webserver/static/js/src/RecentListens.test.tsx

```diff
@@ -813,3 +830,33 @@ describe("Pagination", () => {
     });
   });
 });
+
+describe("pinRecordingModal", () => {
+  it("renders the PinRecordingModal component with the correct props", async () => {
+    const wrapper = mount<RecentListens>(
+      <GlobalAppContext.Provider value={mountOptions.context}>
+        <RecentListens {...props} />
+      </GlobalAppContext.Provider>
+    );
+    const instance = wrapper.instance();
+    const recordingToPin = props.listens[0];
+    let pinRecordingModal = wrapper.find(PinRecordingModal).first();
+
+    // recordingToPin is initially undefined
+    expect(pinRecordingModal.props()).toEqual({
```

Comment:
For my understanding, why are we supporting an undefined `recordingToPin` in the modal? Ideally if we do not have a `recordingToPin` we wouldn't render the modal at all. This would help because then we don't need to care about weird edgecases where recordingToPin is undefined and we're rendering the modal without track names or something. Ideally, we should make the types as strict as possible.

Reply:
The `PinnedRecordingModal ` would have to be rendered before it gets shown: if we only render `PinnedRecordingModal `when `recordingToPin `is set the dataTarget in `ListenControl` isn't able to locate the element with `id=#PinRecordingModal` to show the first time it gets clicked since setState doesn't update `recordingToPin` in the state immediately. 

I changed the component so recordingToPin is initialized to a random listen so we can avoid the weird undefined edgecases like you said.